### PR TITLE
DDL.sql update

### PR DESCRIPTION
**Pull request by Alanasdw**
The database created should now be with UTF8 as the encoding method, this will change all of the nvarchar types to varchar since nvarchar will only use UTF16 and will need some kind of encoding changes outside of the database queries.

Also added the needed primary and foreign key constrains to the tables.

### CODE REVIEW

**Comment by littlepenguin89106** [comment]
primary key的命名慣例基本不會帶有primary key的組成。
其餘命名規格請參照:
[link](https://www.c-sharpcorner.com/UploadFile/f0b2ed/what-is-naming-convention/)

**Comment by littlepenguin89106** [comment]
varchar會造成中英文混合輸入解讀錯誤，建議一律用nvarchar儲存。
請參照:[link](https://ithelp.ithome.com.tw/articles/10213922)

File: DDL.sql [hunk 1]
```diff
@@ -79,8 +80,8 @@ CONSTRAINT FK_tag_value_problem_id FOREIGN KEY ( problem_id) REFERENCES dbo.prob
 CREATE TABLE dbo.account
 (
 user_id         INT             NOT NULL,
-account_name    NVARCHAR( 4000) NOT NULL,
-email           NVARCHAR( 4000) NOT NULL,
+account_name    VARCHAR( 4000) NOT NULL,
+email           VARCHAR( 4000) NOT NULL,
 password        VARCHAR( 200)   NOT NULL,
```
Comment (littlepenguin89106) [inline DDL.sql]: field和設計規格不符

File: DDL.sql [hunk 2]
```diff
@@ -94,7 +95,7 @@ admin_id    INT             NOT NULL,
 user_id     INT             NOT NULL,
 old_auth    INT             NOT NULL,
 new_auth    INT             NOT NULL,
```
Comment (littlepenguin89106) [inline DDL.sql]: 請同步跟新ER diagram的設計疏失

File: DDL.sql [hunk 3]
```diff
@@ -15,66 +16,90 @@ GO
...
-user_id int,
-account_name nvarchar( 4000),
-email nvarchar( 4000),
-passwor varchar( 200),
-auth int
-);
+user_id         INT             NOT NULL,
+account_name    VARCHAR( 4000) NOT NULL,
+email           VARCHAR( 4000) NOT NULL,
+password        VARCHAR( 200)   NOT NULL,
```
Comment (littlepenguin89106) [inline DDL.sql:85]: 與ER diagram設計不符。

File: DDL.sql [hunk 4]
```diff
@@ -15,66 +16,90 @@ GO
...
-admin_id int,
-user_id int,
-old_auth int,
-new_auth int,
-reason nvarchar( 4000)
+datetime    datetime        NOT NULL,
+admin_id    INT             NOT NULL,
+user_id     INT             NOT NULL,
+old_auth    INT             NOT NULL,
+new_auth    INT             NOT NULL,
```
Comment (littlepenguin89106) [inline DDL.sql:97]: 與ER diagram設計不符。

**Review by littlepenguin89106 — COMMENTED** [review]
請更新ER diagram或修正程式碼錯誤

File: DDL.sql (hunk 4 above)
Comment (Alanasdw) [reply DDL.sql:97]: changed ER diagram to be the same with the code

File: DDL.sql (hunk 3 above)
Comment (Alanasdw) [reply DDL.sql:85]: Does the account name need to be there? This is a major issue that needs confirmation.
email and password will be patched accordingly

**Comment by Alanasdw** [comment]
currently the naming convention will be according to [general database naming convention](https://www.c-sharpcorner.com/UploadFile/f0b2ed/what-is-naming-convention/) and [foreign key for two columns](https://dba.stackexchange.com/questions/105251/is-there-a-standard-for-naming-foreign-keys-when-two-columns-reference-the-same)

Another point is that there will need to be some confirmation on the necessity of account.account_name, and the encoding of the password.

The encoding for UTF8 is currently behaving and not cause problems on my local test server, but the caveat is that the supported version is SQL Server 2019 at minimum, which is fairly new. In case there is any version issue, it will need to change back to UTF16 encoding.

Also, the non-clustered indexes are not in the current version since there is no testing.
Some test data to be put inside the database is urgent.

File: DDL.sql [hunk 5]
```diff
@@ -1,80 +1,105 @@
...
 
-CREATE TABLE dbo.answer
+CREATE TABLE dbo.problems
 (
-answer_id bigint,
-problem_id varchar( 30),
-survey_id int,
-answer nvarchar( 4000)
+problem_id      VARCHAR( 30)    NOT NULL,
+topic           VARCHAR( 4000) NOT NULL,
```
Comment (littlepenguin89106) [inline DDL.sql:40]: 我跟改的ER diagram中與此對應的field，僅留下topic作為唯一英文對應，避免混淆

**Review by littlepenguin89106 — APPROVED** [review]

**Pull request merged**